internal/cmp/normalize_test.go

```diff
@@ -26,6 +26,45 @@ func TestEndlessRecursion(t *testing.T) {
 				Slice: []string{"A", "B", "C", "sort"},
 			},
 		},
+		{
+			name: "settable and addressable nil slice",
+			data: &struct {
+				Slice []string
+			}{
+				Slice: []string{},
+			},
+			want: &struct {
+				Slice []string
+			}{
+				Slice: nil,
+			},
+		},
+		{
+			name: "unsettable nil slice",
+			data: &struct {
+				slice []string
+			}{
+				slice: []string{},
+			},
+			want: &struct {
+				slice []string
+			}{
+				slice: []string{},
+			},
+		},
+		{
+			name: "unaddressable nil slice",
+			data: struct {
+				Slice []string
+			}{
+				Slice: []string{},
+			},
+			want: struct {
+				Slice []string
+			}{
+				Slice: []string{},
+			},
+		},
```

Comment:
How is this case different from `"settable and addressable nil slice"` ?

Reply:
ah wait, this is an inline struct, not a pointer.

Reply:
`struct` vs `&struct`